### ISSUE TITLE
Filter drawer points to the wrong table when duplicating table blocks / data providers

### ISSUE THREAD

**Issue by melohagan** [Budibase/budibase#6905]
**Hosting**
Self
  - Method: docker compose
  - Budibase Version: 1.1.7
  - App Version: 1.1.7

**Describe the bug**
When you add a filter to a data provider or table block, and then duplicate the data provider, the filter drawer will only show the last data providers filters. 

**To Reproduce**
1. Add a table block with a filter on one value
2. Duplicate that table block and change the filter value
3. Click define filters on the original table block
4. Notice that it shows the filter of the second instead

**Expected behavior**
When I duplicate table blocks / data providers, the filters should be untied / unique to each table block.

**Screenshots**
![filterbug](https://user-images.githubusercontent.com/101575380/181061300-5419684b-5786-4cc3-801d-63830b69af1f.gif)

**Comment by aptkingston** [comment]
Hey @melohagan. Pretty sure this is just an issue with not remounting settings properly. I reckon if you refresh your page, you'll find that the filters actually are unique. I think we're basically just showing the same filter drawer here. Selecting a different component (that isn't a table block) before selecting the other block would probably fix this.

I'll put in a proper remounting fix though.

**Comment by melohagan** [comment]
Selecting a different component helps, although you have to do it every single time you want to switch between filters of the same component type. 
E.g. Edit Table Block A filters, click container, Edit Table Block B filters.

**Comment by aptkingston** [comment]
Yep that just confirms that the bug is what I thought it was - I don't mean swapping components is a valid solution long term haha. I've opened a PR to fix it properly.
